core/framework/extractors/tree_sitter_extractor_backup_20250913.py

```
@@ -0,0 +1,945 @@
+            structure = self._extract_structure(tree, content, language)
+
+            return {
+                'symbols': symbols,
+                'metrics': metrics,
+                'structure': structure,
+                'language': language,
+                'tree_sitter_version': tree_sitter.__version__ if hasattr(tree_sitter, '__version__') else 'unknown'
+            }
+
+        except Exception as e:
+            logger.error(f"Failed to parse {file_path} with Tree-sitter: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)